packages/react/src/components/DataTable/TableToolbarSearch.js

```diff
@@ -57,15 +57,12 @@ const TableToolbarSearch = ({
     }
   }, [focusTarget]);
 
-  useEffect(
-    () => {
-      if (defaultValue) {
-        onChangeProp('', defaultValue);
-      }
-    },
-    //eslint-disable-next-line react-hooks/exhaustive-deps
-    []
-  );
+  useEffect(() => {
```

Comment:
I think the pattern that we use for derived state matches this one from the FAQ: https://reactjs.org/docs/hooks-faq.html#how-do-i-implement-getderivedstatefromprops

We've run into some issues otherwise 😬  (typically infinite render loops with effects)

Might help to instead add a state value for `defaultValue`:

```js
const [prevDefaultValue, setPrevDefaultValue] = useState(defaultValue);

if (prevDefaultValue !== defaultValue) {
  setValue(defaultValue);
  setPrevDefaultValue(defaultValue);
}
```

Similarly, calling `onChange` during these updates has been something that we've noticed is easy to trigger infinite loops (or bad state updates) so we tend to avoid it in these use-cases.

Reply:
Hey yeah this makes sense, thanks for the response. In our case it's a bit more complicated as we have a reducer keeping track of a large state object but I effectively ended up doing the same as the snippet and it's working as intended 👍  Going to close this out. Thanks for the feedback/help!